app/services/autograntable.rb

```diff
@@ -0,0 +1,84 @@
+class Autograntable
```

Comment:
I really like this code. Really clear, nicely divided up into sensible chunks. Big thumbs up from me.